sdk/ml/azure-ai-ml/azure/ai/ml/entities/_util.py

```diff
@@ -191,21 +191,36 @@ def validate_attribute_type(attrs_to_check: dict, attr_type_map: dict):
                 error_type=ValidationErrorType.INVALID_VALUE,
             )
 
+def is_empty_target(obj):
+    """Determines if it's empty target"""
+    return (obj is None
+            # some objs have overloaded "==" and will cause error. e.g CommandComponent obj
+            or (isinstance(obj, dict) and len(obj) == 0)
+        )
 
-def convert_ordered_dict_to_dict(target_object: Union[Dict, List]) -> Union[Dict, List]:
-    """Convert ordered dict to dict.
+def convert_ordered_dict_to_dict(target_object: Union[Dict, List], remove_empty=True) -> Union[Dict, List]:
+    """Convert ordered dict to dict. Remove keys with None value.
 
     This is a workaround for rest request must be in dict instead of
     ordered dict.
     """
     # OrderedDict can appear nested in a list
     if isinstance(target_object, list):
-        target_object = [convert_ordered_dict_to_dict(obj) for obj in target_object]
-    if isinstance(target_object, dict):
-        for key, dict_candidate in target_object.items():
-            target_object[key] = convert_ordered_dict_to_dict(dict_candidate)
+        new_list = []
+        for item in target_object:
+            item = convert_ordered_dict_to_dict(item)
+            if not is_empty_target(item) or not remove_empty:
+                new_list.append(item)
+        return new_list
     if isinstance(target_object, OrderedDict):
```

Comment:
is this line still needed? Is OrderedDict a subclass of dict?

Reply:
It is indeed a subclass of dict, should work